### PR TITLE
[Docs] Add more docs to the sse4.2 cmpstr fns

### PR DESCRIPTION
**Pull request by dlrobertson**
- Add more examples to `_mm_cmpistri`
 - Add basic docs to `_mm_cmpestri`
 - Cleanup lib docs

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -67,32 +67,28 @@
...
-//!         }
-//!
-//!         and_256();
-//!     } else {
-//!         println!("avx2 intrinsics will not work, they may generate SIGILL");
-//!     }
+//! # #![feature(cfg_target_feature)]
+//! # #![feature(target_feature)]
+//! # #[macro_use]
+//! # extern crate stdsimd;
```
Comment (dlrobertson) [inline src/lib.rs]: Realized this section was necessary to show, since the previous example shows this

File: src/x86/sse42.rs [hunk 2]
```diff
@@ -255,7 +343,87 @@ pub unsafe fn _mm_cmpestrm(
...
-/// control in `imm8`, and return the generated index.
+/// control in `imm8`, and return the generated index. Similar to [`_mm_cmpistri`]
+/// with the excception that [`_mm_cmpistri`] implicityly determines the length of
+/// `a` and `b`.
+///
+/// # Control modes
+///
+/// The control specified by `imm8` may be one or more of the following.
+///
+/// ## Data size and signedness
```
Comment (dlrobertson) [inline src/x86/sse42.rs]: I don't like duplicating the `mode` section

File: src/x86/sse42.rs [hunk 3]
```diff
@@ -255,7 +343,87 @@ pub unsafe fn _mm_cmpestrm(
 }
 
 /// Compare packed strings `a` and `b` with lengths `la` and `lb` using the
-/// control in `imm8`, and return the generated index.
+/// control in `imm8`, and return the generated index. Similar to [`_mm_cmpistri`]
+/// with the excception that [`_mm_cmpistri`] implicityly determines the length of
```
Comment (dlrobertson) [inline src/x86/sse42.rs]: I thought I would document what is unique about `_mm_cmpestri` and then add links to `_mm_cmpistri` where I have more examples

**Comment by alexcrichton** [comment]
Looks great to me, thanks @dlrobertson! The duplication I think is ok for now, although we can trim down later if we feel necessary.

**Pull request merged**